e2e/app/context.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ramendr/ramen/e2e/config"
+	"github.com/ramendr/ramen/e2e/types"
+)
+
+// Context implements types.Context for sharing the log, env, and config with all code.
+type BaseContext struct {
+	log    *zap.SugaredLogger
+	env    *types.Env
+	config *config.Config
+	ctx    context.Context
+}
+
+func (c *BaseContext) Logger() *zap.SugaredLogger {
+	return c.log
+}
+
+func (c *BaseContext) Config() *config.Config {
+	return c.config
+}
+
+func (c *BaseContext) Env() *types.Env {
+	return c.env
+}
+
+func (c *BaseContext) Context() context.Context {
+	return c.ctx
+}
+
+// WithTimeout returns a derived context with a deadline. Call cancel to release resources associated with the context
+// as soon as the operation running in the context complete.
+func (c BaseContext) WithTimeout(d time.Duration) (*BaseContext, context.CancelFunc) {
+	ctx, cancel := context.WithTimeout(c.ctx, d)
+	c.ctx = ctx //nolint:revive
+
+	return &c, cancel
+}
```

Comment:
The rest of the function are not needed or wanted. Context should be ready only - once we created it, you cannot change it.